src/stan/callbacks/writer.hpp

```diff
@@ -46,6 +46,12 @@ class writer {
    */
   virtual void operator()(const std::string& message) {}
 
+  /**
+   * Checks if stream is valid.
+   */
+  virtual bool is_nonnull() const noexcept { return false; }
+  virtual const char* comment_prefix() const noexcept { return "# "; }
```

Comment:
I don't like making this part of the writer API, especially if we ever hope to have non-csv-based IO that may not have a prefixed-comment style

What problem is it solving?